cogs/score.py

```diff
@@ -42,146 +43,217 @@ async def on_ready(self):
 
     @commands.Cog.listener()
-    async def on_member_update(self, before, after):
-        if before.display_name == after.display_name:
+    async def on_user_update(self, before, after):
+        if str(before) == str(after):
             return
 
-        m_id = after.id
-        new_nick = after.display_name
+        user_id = after.id
+        new_name = str(after)
         conn = sqlite3.connect(self.bot.config.db_path)
         c = conn.cursor()
-
-        if not c.execute("SELECT * FROM Members WHERE ID = ?", (m_id, )):
-            c.execute("INSERT INTO Members VALUES (?, ?, ?)", (
-                m_id,
-                new_nick,
-                0,
-            ))
-            conn.commit()
-
-        else:
-            c.execute("UPDATE Members SET DisplayName = ? WHERE ID = ?", (
-                new_nick,
-                m_id,
-            ))
-            conn.commit()
-
+        c.execute('INSERT OR REPLACE INTO Members VALUES (?,?)',
+                  (user_id, new_name))
+        conn.commit()
         conn.close()
 
     @commands.command()
-    async def ranking(self, ctx):
+    async def score(self, ctx, str1: str = None, *, str2: str = None):
+        """Display your score for a given emoji. If no arguments are given,
+           displays your total upmartlet/downmartlet score.
+           Arguments: An emoji (optional) and a mention (optional), in any order
         """
-        Upmartlet Rankings! :^)
-        """
-        await ctx.trigger_typing()
+        mentions = ctx.message.mentions
+        member = ctx.message.author
+        emoji = None
+
+        if mentions:
+            member = mentions[0]
+            if str1 and str1.replace("!", "") != mentions[0].mention.replace(
+                    "!", ""):
+                emoji = str1
+
+            if str2 and str2.replace("!", "") != mentions[0].mention.replace(
+                    "!", ""):
+                emoji = str2
+        elif str1:
+            emoji = str1
+
+        if emoji:
+            try:
+                await ctx.message.remove_reaction(emoji, self.bot.user)
+            except discord.errors.HTTPException:
+                await ctx.send("Invalid emoji or username")
+                return
+
+        m_id = member.id
+        nick = member.display_name
 
         conn = sqlite3.connect(self.bot.config.db_path)
         c = conn.cursor()
-        c.execute("SELECT * FROM Members ORDER BY Score DESC;")
-        members = c.fetchall()
-
-        if not members:
-            await ctx.send(
-                "Ranking is not yet available for this server, please "
-                "upvote/downvote moar.")
-            return
 
-        table = []
-        table_list = []
-        counter = 1
+        if not emoji:
+            emoji = "total"
+            c.execute(
+                "SELECT count(ReacteeID) FROM Reactions WHERE ReacteeID = ? AND ReactionName = ? AND ReacterID != ReacteeID",
```

Comment:
this is over 80 chars, im surprised YAPF didnt yell

Reply:
oops didn't notice, and yeah I think yapf doesn't yell for strings

Reply:
Resolved in 0847cae